### PR TITLE
Added EfficientNet lite option.

### PR DESCRIPTION
**Pull request by sebastian-sz**
# What does this PR do

This PR adds `lite` option to existing EfficientNet V1's.

EfficientNet lite's are slightly modified efficientnets that better support edge devices. 

The models and weights are ported from original repository:
https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/lite/README.md 

### Linked issues:
In Keras repo: #16835  
Previously in Tensorflow repo: 
https://github.com/tensorflow/tensorflow/pull/48400
https://github.com/tensorflow/tensorflow/issues/45091

# How does this affect the API
This works similar to `minimalistic` flag in MobileNetV3. Users can create EfficientNet lite's by passing `lite=True` argument upon model creation:
```python
model = tf.keras.applications.efficientnet.EfficientNetB0(lite=True)
```
(Keep in mind that the above, will raise an error as the weights are not in the GCS bucket. The above works with `weights=None`)

NOTE: lite options have been only released for B0-B4 variants.

# Pretrained weights
The weights have been converted from the original repository, using already present [efficientnet_weight_update_util.py](https://github.com/keras-team/keras/blob/master/keras/applications/efficientnet_weight_update_util.py).

I cannot attach them to this PR as they exceed the allowed file size. They can be downloaded from my Google Drive:
https://drive.google.com/file/d/1XSpE1NGhbaP8dN1Rem3QrmeuC--2E9QR/view?usp=sharing 

# Imagenet Accuracy
I evaluated the models using my [external repository](https://github.com/sebastian-sz/efficientnet-lite-keras/tree/main/imagenet_evaluation), so I'm copy pasting the accuracy results below:
#### Results table
| Variant | Image size | Reported Top 1 | This repo Top 1 | This repo Top 5 | Top 1 difference |
| ------- | ---------- | -------------- | --------------- | --------------- | ---------------- |
|   B0    |     224    |     75.1       |       75.1      |       92.3      |       0.0        |
|   B1    |     240    |     76.7       |       76.8      |       93.3      |       +0.1       |
|   B2    |     260    |     77.6       |       77.6      |       93.8      |       0.0        |
|   B3    |     280    |     79.8       |       79.4*     |       94.7      |       -0.4      |
|   B4    |     300    |     81.5       |       80.8*     |       95.2      |       -0.7      |

We can observe that the accuracy for B3/B4 variants slightly differs from reported. The errors are below 1% and the model outputs the same probabilities as the ones in the original repository.

If this is an issue we can discuss the origin of the differences - let me know.

# Known Issues
Similar to `minimalistic` MobileNet's V3 the models are not covered under `applications_test.py`, because they are created based on a parameter. It would be nice to have those tests, I can try to further modify `applications_test.py` to try and include these models. Let me know.

### CODE REVIEW

File: keras/applications/efficientnet.py [hunk 1]
```diff
@@ -354,23 +389,31 @@ def round_repeats(repeats):
 
     # Build stem
     x = img_input
-    x = layers.Rescaling(1.0 / 255.0)(x)
-    x = layers.Normalization(axis=bn_axis)(x)
-    if weights == "imagenet":
-        # Note that the normaliztion layer uses square value of STDDEV as the
+
+    if lite:
+        x = layers.Normalization(mean=127.0, variance=128.0**2)(x)
```
Comment (fchollet) [inline keras/applications/efficientnet.py:394]: We should use the same preprocessing as the below, for consistency. This means that the Normalization layer should be adapted to account for the scaling, and the normalization state should be part of the weights file.

File: keras/applications/efficientnet.py [hunk 2]
```diff
@@ -202,6 +223,10 @@
     input_shape: Optional shape tuple, only to be specified
         if `include_top` is False.
         It should have exactly 3 inputs channels.
+    lite: Optional boolean, whether to create a 'lite' variant of the network.
+        Note, that only B0-B4 variants have lite option. For more details on
+        lite variants, you can read:
+        https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/lite/README.md
```
Comment (fchollet) [inline keras/applications/efficientnet.py:229]: They should be separate classes each. This will also make them testable.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! This is great.

File: keras/applications/efficientnet.py (hunk 1 above)
Comment (sebastian-sz) [reply keras/applications/efficientnet.py:394]: @fchollet 
Given these models will be in a separate class / file I was thinking about adding `include_preprocessing` option, as it is in newer implementations, like EfficientNetV2 or ResNet-RS.

Putting the Normalization parameters will be incompatible with this option as the `Normalization` layer will always have to be present. 

Perhaps instead of putting the normalization parameters in the weights file, the `include_preprocessing` option could be used?

Please let me know!

File: keras/applications/efficientnet.py (hunk 2 above)
Comment (sebastian-sz) [reply keras/applications/efficientnet.py]: @fchollet Moved the models into a separate file.

**Comment by sebastian-sz** [comment]
I might have accidentally removed the request for review from @haifeng-jin - I did not mean to do that.

File: keras/applications/efficientnet.py (hunk 1 above)
Comment (fchollet) [reply keras/applications/efficientnet.py]: > Perhaps instead of putting the normalization parameters in the weights file, the include_preprocessing option could be used?

Sounds good to me!

**Comment by tanzhenyu** [comment]
Re Pretrained Weights -- do you have a colab that can help us evaluate the model using the weights from Google Drive?

**Comment by sebastian-sz** [comment]
@tanzhenyu Here is a [Colab link](https://colab.research.google.com/drive/1ZxiUKAgdyK_qAspgouNRjiaKHAa_zgdw?usp=sharing)

Few notes:
* The evaluation requires TFRecords created via[ scripts in tensorflow/tpu repository](https://github.com/tensorflow/tpu/tree/master/tools/datasets#imagenet_to_gcspy).
* The colab is mostly notebook version of eval script from [my repo](https://github.com/sebastian-sz/efficientnet-lite-keras/tree/main/imagenet_evaluation)

Let me know if this works for you!

File: keras/applications/efficientnet.py (hunk 1 above)
Comment (sebastian-sz) [reply keras/applications/efficientnet.py]: Added `include_preprocessing` option.

**Comment by tanzhenyu** [comment]
> @tanzhenyu Here is a [Colab link](https://colab.research.google.com/drive/1ZxiUKAgdyK_qAspgouNRjiaKHAa_zgdw?usp=sharing)
> 
> Few notes:
> 
> * The evaluation requires TFRecords created via[ scripts in tensorflow/tpu repository](https://github.com/tensorflow/tpu/tree/master/tools/datasets#imagenet_to_gcspy).
> * The colab is mostly notebook version of eval script from [my repo](https://github.com/sebastian-sz/efficientnet-lite-keras/tree/main/imagenet_evaluation)
> 
> Let me know if this works for you!

Looks good to me!

**Comment by LukeWood** [comment]
@sebastian-sz can we move this PR to KerasCV?

**Comment by sebastian-sz** [comment]
@LukeWood yes, I will try to port this PR until the end of the month. I will close this PR once I port it to keras-cv (unless you want to close now).

**Comment by LukeWood** [comment]
> @LukeWood yes, I will try to port this PR until the end of the month. I will close this PR once I port it to keras-cv (unless you want to close now).

Thank you!  Keep me posted, feel free to request review.

**Comment by sebastian-sz** [comment]
This PR has been moved to `keras-cv`. Closing this for now.
